htdocs/xoops_lib/Xoops/Core/Events.php

```diff
@@ -211,6 +211,19 @@ public function addListener($eventName, $callback)
         $eventName = $this->toInternalEventName($eventName);
         $this->eventListeners[$eventName][]=$callback;
     }
+    
+    /**
+     * removeListener - remove a listener.
+     *
+     * @param string   $eventName the event name
+     *
+     * @return void
+     */
+    public function removeListener($eventName)
```

Comment:
No. This could wreak havoc. There is no safe way to permit this.

An added listener does not persist anyway, so this shouldn't be needed.